addons/mail/models/discuss/discuss_channel.py

```diff
@@ -392,29 +392,33 @@ def get_field_name(field_description):
 
         def get_vals(channel):
             return {
-                get_field_name(field_description): (
-                    channel[get_field_name(field_description)],
-                    field_description,
-                )
-                for field_description in self._sync_field_names()
+                subchannel: {
+                    get_field_name(field_description): (
+                        channel[get_field_name(field_description)],
+                        field_description,
+                    )
+                    for field_description in field_descriptions
+                }
+                for subchannel, field_descriptions in self._sync_field_names().items()
             }
 
         old_vals = {channel: get_vals(channel) for channel in self}
         result = super().write(vals)
         for channel in self:
-            new_vals = get_vals(channel)
-            diff = []
-            for field_name, (value, field_description) in new_vals.items():
-                if value != old_vals[channel][field_name][0]:
-                    diff.append(field_description)
-            if diff:
-                channel._bus_send_store(channel, diff)
+            new_subchannel_vals = get_vals(channel)
+            for subchannel, vals in new_subchannel_vals.items():
+                diff = []
+                for field_name, (value, field_description) in vals.items():
+                    if value != old_vals[channel][subchannel][field_name][0]:
+                        diff.append(field_description)
+                if diff:
+                    channel._bus_send_store(channel, diff, subchannel=subchannel)
```

Comment:
It would be nice to have a test in livechat with the note field to cover this change (checking bus table) as it's a bit tricky and won't be covered by existing tests.